sdk/storage/Azure.Storage.Common/tests/StorageResponseClassifierTests.cs

```diff
@@ -70,6 +70,27 @@ public void IsRetriableResponse_StorageErrors_SecondaryUri(string errorCode)
             Assert.IsTrue(classifier.IsRetriableResponse(message));
         }
 
+        [TestCase("ContainerAlreadyExists", "If-Match", false)]
+        [TestCase("ContainerAlreadyExists","If-None-Match", false)]
+        [TestCase("ContainerAlreadyExists","If-Unmodified-Since", false)]
+        [TestCase("ContainerAlreadyExists","If-Modified-Since", false)]
+        [TestCase("BlobAlreadyExists", "If-Match", false)]
+        [TestCase("BlobAlreadyExists", "If-None-Match", false)]
+        [TestCase("BlobAlreadyExists", "If-Unmodified-Since", false)]
+        [TestCase("BlobAlreadyExists", "If-Modified-Since", false)]
+        [TestCase("InternalError", "If-Match", true)]
+        [TestCase("BlobAlreadyExists", "Non-Conditional-Header", true)]
+        public void IsError_409_ConditionalResponse(string code, string header, bool isError)
+        {
+            var mockRequest = new MockRequest();
+            mockRequest.Headers.Add("x-ms-error-code", code);
```

Comment:
This test adds the error code to the request, when it should be on the response.